foursight_core/app_utils.py

```diff
@@ -131,6 +155,78 @@ def init_response(self, environ):
             response.status_code = 400
         return connection, response
 
+    def is_running_locally(self, request_dict):
+        return request_dict.get('context', {}).get('identity', {}).get('sourceIp', '') == "127.0.0.1"
+
+    def get_logged_in_user_info_from_jwt_token_cookie(self, request_dict):
+        email_address = ""
+        email_verified = ""
+        first_name = ""
+        last_name = ""
+        issuer = ""
+        subject = ""
+        audience = ""
+        issued_time = ""
+        expiration_time = ""
+        jwt_decoded = ""
+        try:
+            jwt_token = self.get_jwt(request_dict)
+            if jwt_token:
+                options = {"verify_signature": False, "verify_aud": False}
+                algorithms = ["HS256"]
+                jwt_decoded = jwt.decode(jwt_token, 'secret', algorithms=algorithms, options=options)
+                if jwt_decoded:
+                    email_address = jwt_decoded.get("email")
+                    email_verified = jwt_decoded.get("email_verified")
+                    issuer = jwt_decoded.get("iss")
+                    if issuer:
+                        name = jwt_decoded.get(issuer + "name")
+                        if name:
+                            first_name = name.get("name_first")
+                            last_name = name.get("name_last")
+                    subject = jwt_decoded.get("sub")
+                    audience = jwt_decoded.get("aud")
+                    issued_time = self.convert_time_t_to_useastern_datetime(jwt_decoded.get("iat"))
+                    expiration_time = self.convert_time_t_to_useastern_datetime(jwt_decoded.get("exp"))
+        except Exception as e:
+            logger.warn("Cannot get logged in user info from JWT token (not fatal - just for Foursight UI display).")
+            logger.warn(e)
+        return {"email_address": email_address,
+                "email_verified": email_verified,
+                "first_name": first_name,
+                "last_name": last_name,
+                "issuer": issuer,
+                "subject": subject,
+                "audience": audience,
+                "issued_time": issued_time,
+                "expiration_time": expiration_time,
+                "jwt": jwt_decoded}
+
+    def get_portal_url(self, env_name: str, stage_name: str) -> str:
+        if not self.portal_url:
+            try:
+                environment_and_bucket_info = self.environment.get_environment_and_bucket_info(env_name, stage_name)
+                self.portal_url = environment_and_bucket_info.get("fourfront")
+            except Exception as e:
+                logger.error(f"Error determining portal URL")
```

Comment:
So this is an error, not a warning, yet you're still trapping it and making do. Something seems odd about that.

Reply:
Yeah, mistake, this is a hard error because we now use this (portal URL) to get the Auth0 client ID (previously hardcoded value). Letting it raise the exception now.